### PR TITLE
Implement basic logging functionality

### PR DESCRIPTION
**Pull request by celery6**
some notes
- only 1 log function can run at once, so logging is blocking up to 50 ticks (arbitrary number for now)
  - logs can be printed out of order cuz tasks are unblocked by priority not time. however the timestamp is acquired before waiting, so actual order can be post-processed if needed
- still needs more thinking about full buffer edge cases for protecting against out of bounds and overwriting !!!!!!!! Are there any glaring issues currently?
- since the logger function uses snprintf, all tasks that log will get stack eaten by snprintf :( should do better stack usage analysis someday to make sure this is fine...
- not sure if each buffer itself should be protected by a mutex or if the isFull flag is enough. I think using per-buffer mutexes would still require the isFull flag due to the handoff between log funcs and dump queue (something needs to guard a full buffer while its waiting to be dumped). idk

### CODE REVIEW

**Comment by Joe-Joe-Joe-Joe** [comment]
Have you tested on the hardware yet?

File: STM32Cube/Tasks/log.c [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+void logTask(void *argument)
+{
+    log_buffer* bufferToPrint;
+
+    // wait for a full buffer to appear in the queue; timeout is long - queues are not expected to fill up super quickly
+    for (;;)
+    {
+		if (xQueueReceive(fullBuffersQueue, &bufferToPrint, 1000000) == pdPASS)
+        {
+            // TODO: do uart transmit better (use _IT?)
```
Comment (Joe-Joe-Joe-Joe) [inline STM32Cube/Tasks/log.c:184]: *Use DMA

**Review by Joe-Joe-Joe-Joe — CHANGES_REQUESTED** [review]
See comments. Also test the code if that hasn't already been done. Design wise I'm very happy with it for now. Stuff like efficiency improvements on the UART is not necessary for now, but quickly will be once we want to run threads at any reasonable speed (no 1 second timeouts lol).

**Comment by celery6** [comment]
Tested latest changes, works well. Tested spamming logs in all tasks and buffers don't get full with the current osDelay(1000), but they do get full with shorter osDelays. To be optimized later ig!

merged master but didn't initialize task in main.

**Comment by celery6** [comment]
Yes

**Comment by Joe-Joe-Joe-Joe** [comment]
try bumping baud rate to 500000
I know for a fact the chip supports it, idk if the programmer will bottleneck us there tho

**Review by Joe-Joe-Joe-Joe — APPROVED** [review]

**Pull request merged**